terraform/modules/aws-network-load-balancer/resources.lb.tf

```diff
@@ -0,0 +1,68 @@
+resource "aws_lb" "nlb" {
+  name = "${var.environment}-loadbalancer"
+
+  internal           = false
+  load_balancer_type = "network"
+  enable_cross_zone_load_balancing = true
+
+  subnets = data.aws_subnet_ids.public_subnets.ids
+
+  tags = {
+    Environment = var.environment
+  }
+}
+
+
+resource "aws_lb_listener" "ingress" {
```

Comment:
Make this an explicit listener for http, and one for https.

I assume we want to do HTTPS everywhere anyways, so we probably can also just make the http lb listener emit the redirect and not attach any machines, and add a target group to the https listener.

Reply:
Since it's a Network LB, I think this layer 7 behavior is better suited to be handled by whatever sits behind the LB. The context of k8s masters, probably makes this approach again appear in a more reasonable light, doesn't it?

Reply:
Yeah, fine for me - so let's just "write down" two individual `aws_lb_listener` `ingress_http` and `ingress_https` resources.